core-plugins/src/main/java/co/cask/hydrator/plugin/common/KinesisUtil.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.common;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.model.CreateStreamRequest;
+import com.amazonaws.services.kinesis.model.DescribeStreamRequest;
+import com.amazonaws.services.kinesis.model.ResourceNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility methods to help create and manage Kinesis Streams
+ */
+public final class KinesisUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KinesisUtil.class);
+  private static final long DELETE_TIMEOUT = 1000 * 120;
+  private static final long CREATE_TIMEOUT = 3 * 60 * 1000;
+  private static final long SLEEP_INTERVAL = 1000 * 10;
+
+  private static final String STATUS_CREATING = "CREATING";
+  private static final String STATUS_DELETING = "DELETING";
+  private static final String STATUS_ACTIVE = "ACTIVE";
+  private static final String STATUS_UPDATING = "UPDATING";
+
+
+  private KinesisUtil() {
+  }
+
+  /**
+   * Creates an Amazon Kinesis stream if it does not exist and waits for it to become available
+   *
+   * @param kinesisClient The {@link AmazonKinesisClient} with Amazon Kinesis read and write privileges
+   * @param streamName The Amazon Kinesis stream name to create
+   * @param shardCount The shard count to create the stream with
+   * @throws IllegalStateException Invalid Amazon Kinesis stream state
+   * @throws IllegalStateException Stream does not become active before the timeout
+   */
+  public static void createAndWaitForStream(AmazonKinesisClient kinesisClient,
+                                            String streamName,
+                                            int shardCount) {
+    if (streamExists(kinesisClient, streamName)) {
+      String state = getStreamState(kinesisClient, streamName);
+      switch (state) {
+        case STATUS_DELETING:
+          long waitTimeDelete = System.currentTimeMillis() + DELETE_TIMEOUT;
```

Comment:
Instead of calling ``System.currentTimeMillis()`` twice, which could return two different values, extract it to a variable ``deleteStartTime`` and use it?